### PR TITLE
Fix ambient lighting not showing on Transparent surfaces

### PR DESCRIPTION
**Pull request by samcake**
https://highfidelity.fogbugz.com/f/cases/edit/15104/Materials-with-transparency-don-t-receive-reflections-from-the-ambient-map-in-RC67

This PR fixes this bug ^.
the problem was that the ambient map & buffer wasn't bound correctly during the rendering of the transparent objects.
The ambient lighting on transparent objects wasn't displayed correctly.
To address this, we made sure the place where the shader slots are defined for the global lighting &  shading resources are unique accross the different pass of the shaders.

While doing this work and in order to help with debugging, we added support for disabling enabling the Haze effect and the Emissive effect from the Lighting Model and from the deferredREndering.qml ui tool for Luci.js

TEST PLAN
Run the tests in this folder:
https://github.com/highfidelity/hifi_tests/tree/master/tests/engine/render/material

Going into domain with translucent cmaterials such as Blue or dev-avatarisland validate that the ambient lighting is proeperly applied to the transparent objects
You can do so by running the Luci app and checking the lighting with AMbient only

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/9778/HighFidelity-Beta-PR13170-8f34043bde1fe3ca9b043de30b04b78efbb6a706.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9778/HighFidelity-Beta-PR13170-8f34043bde1fe3ca9b043de30b04b78efbb6a706.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9778/HighFidelity-Beta-Interface-PR13170-8f34043bde1fe3ca9b043de30b04b78efbb6a706.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9778/HighFidelity-Beta-PR13170-8f34043bde1fe3ca9b043de30b04b78efbb6a706.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9778/HighFidelity-Beta-Interface-PR13170-8f34043bde1fe3ca9b043de30b04b78efbb6a706.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/9785/HighFidelity-Beta-PR13170-aee40fbd6c929c1505f200ada00458270271fe5f.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9785/HighFidelity-Beta-PR13170-aee40fbd6c929c1505f200ada00458270271fe5f.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9785/HighFidelity-Beta-Interface-PR13170-aee40fbd6c929c1505f200ada00458270271fe5f.dmg

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/9789/HighFidelity-Beta-PR13170-16c60850ff4afb21650721a945b2f525775237db.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9785/HighFidelity-Beta-PR13170-aee40fbd6c929c1505f200ada00458270271fe5f.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9785/HighFidelity-Beta-Interface-PR13170-aee40fbd6c929c1505f200ada00458270271fe5f.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9789/HighFidelity-Beta-PR13170-16c60850ff4afb21650721a945b2f525775237db.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9789/HighFidelity-Beta-Interface-PR13170-16c60850ff4afb21650721a945b2f525775237db.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9789/HighFidelity-Beta-PR13170-16c60850ff4afb21650721a945b2f525775237db.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9789/HighFidelity-Beta-Interface-PR13170-16c60850ff4afb21650721a945b2f525775237db.exe

File: libraries/render-utils/src/LightingModel.h [hunk 1]
```diff
@@ -86,19 +89,21 @@ class LightingModel {
...
-
         float enableObscurance{ 1.0f };
-
         float enableMaterialTexturing { 1.0f };
         float enableWireframe { 0.0f }; // false by default
 
+        float enableHaze{ 1.0f };
+        float spare1;
+        float spare2;
+        float spare3;
```
Comment (AndrewMeadows) [inline libraries/render-utils/src/LightingModel.h]: Not a blocker but I'm mildly curious as to why the `enableFoo` data members are of type `float` and not `bool`.

I assume the `spareX` data members are for` sizeof` padding.

**Comment by AndrewMeadows** [comment]
The ubuntu build suffered a mysterious failure during the link stage.  Link errors are not evident in the log...perhaps it ran out of memory?  Jenkins, rebuild this please to see if it is reproducible.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/9797/HighFidelity-Beta-PR13170-16c60850ff4afb21650721a945b2f525775237db.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9797/HighFidelity-Beta-PR13170-16c60850ff4afb21650721a945b2f525775237db.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9797/HighFidelity-Beta-Interface-PR13170-16c60850ff4afb21650721a945b2f525775237db.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9797/HighFidelity-Beta-PR13170-16c60850ff4afb21650721a945b2f525775237db.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9797/HighFidelity-Beta-Interface-PR13170-16c60850ff4afb21650721a945b2f525775237db.exe

File: libraries/render-utils/src/RenderDeferredTask.cpp [hunk 2]
```diff
@@ -337,11 +337,18 @@ void DrawDeferred::run(const RenderContextPointer& renderContext, const Inputs&
         // Setup lighting model for all items;
         batch.setUniformBuffer(render::ShapePipeline::Slot::LIGHTING_MODEL, lightingModel->getParametersBuffer());
 
+        // Set the light
+        deferredLightingEffect->setupKeyLightBatch(args, batch,
+            render::ShapePipeline::Slot::KEY_LIGHT,
+            render::ShapePipeline::Slot::LIGHT_AMBIENT_BUFFER,
+            render::ShapePipeline::Slot::LIGHT_AMBIENT_MAP);
```
Comment (SamGondelman) [inline libraries/render-utils/src/RenderDeferredTask.cpp]: If we do this here, do we still need the lightBatchSetter in RenderPipelines.cpp?

**Comment by SamGondelman** [comment]
Please test this on Android, too.

File: libraries/render-utils/src/RenderDeferredTask.cpp (hunk 2 above)
Comment (samcake) [reply libraries/render-utils/src/RenderDeferredTask.cpp]: we cold avoid to do it, i m shy

**Comment by samcake** [comment]
this has been fixed in 67.2 and merged back to master
closing

**Pull request closed without merging**